pkg/util/workloadrepo/snapshot.go

```diff
@@ -218,18 +187,54 @@ func (w *worker) startSnapshot(_ctx context.Context) func() {
 				if err := updateHistSnapshot(ctx, sess, snapID, errs); err != nil {
 					logutil.BgLogger().Info("workload repository snapshot failed: could not update hist_snapshots", zap.NamedError("err", err))
 				}
+
+				continue
+			case <-w.snapshotChan:
+			case <-w.snapshotTicker.C:
+			}
+
+			// coordination logic
+			if !w.owner.IsOwner() {
+				continue
+			}
+
+			for range 5 {
+				snapID, err := w.getSnapID(ctx)
+				if err != nil {
+					logutil.BgLogger().Info("workload repository cannot get current snapid", zap.NamedError("err", err))
+					continue
+				}
+				// use upsert such that this SQL does not fail on duplicated snapID
+				//
```

Comment:
I tried to refine the comment here:

```
// Use UPSERT to ensure this SQL doesn't fail on duplicate snapID.
//
// NOTE: In a highly unlikely corner case, there could be two owners.
// This might occur if upsertHistSnapshot succeeds but updateSnapID fails 
// due to another owner winning the etcd CAS loop.
// While undesirable, this scenario is acceptable since both owners would 
// likely share similar datetime values and same cluster version.
```
PTAL